lib/src/components/NxList/NxList.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, MutableRefObject, useCallback, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import NxListText from './NxListText';
+import NxListSubtext from './NxListSubtext';
+import NxListActions from './NxListActions';
+import NxListButtonItem from './NxListButtonItem';
+import NxListLinkItem from './NxListLinkItem';
+import NxListItem from './NxListItem';
+import NxListDescriptionTerm from './NxListDescriptionTerm';
+import NxListDescription from './NxListDescription';
+import { NxH3 } from '../SimpleComponents';
+import { splitOutFirst } from '../../util/childUtil';
+import { NxListProps, NxLoadError, NxLoadingSpinner } from '../..';
+import useMutationObserver from '@rooks/use-mutation-observer';
+import { NxListComponent, nxListPropTypes } from './types';
+
+const mutationObserverConfig = { subtree: false, childList: true, attributes: false, characterData: false };
+
+const NxList = forwardRef<HTMLUListElement, NxListProps>((props: NxListProps, ref) => {
+  const {className, children, bulleted, emptyMessage, isLoading = false, error, retryHandler, ...attrs } = props;
+  const classNames = classnames(className, 'nx-list', {'nx-list--bulleted': bulleted});
+  const [title, otherChildren] = splitOutFirst(NxH3, children);
```

Comment:
As I mentioned in another comment, I don't think the header should be within the list at all, so this special logic to handle it can go away.

Reply:
> As I mentioned in another comment, I don't think the header should be within the list at all, so this special logic to handle it can go away.

All logic concerning headers have now been removed.
7d05c8feb49640bac834968a1bfd984deb457640